File: 2231-find-first-palindromic-string-in-the-array/find-first-palindromic-string-in-the-array.cpp
class Solution {
private:
 bool firstPalindromeutil(string s)
    {
        int i =0 ,j = s.size()-1;
        while(i<j){
            if(s[i]!=s[j]) 
               return false;
            i++;
            j--;
        }
        return true;
    }
public:
    string firstPalindrome(vector<string>& words) {
        for(int i=0;i<words.size();i++)
           {
              if(firstPalindromeutil(words[i]))
                  return words[i];
           }
        return "";
    }
};

File: 2231-find-first-palindromic-string-in-the-array/find-first-palindromic-string-in-the-array.py
class Solution:
    def firstPalindrome(self, words: List[str]) -> str:
        def isPalindrome(word):
            if word == word[::-1]:
                return True
        
        for c in words:
            if isPalindrome(c):
                return c
            else:
                continue

        return ""


